lib/engine/db/postgres/index.ts

```diff
@@ -0,0 +1,150 @@
+        client.query(string, args, (err, result) => {
+            if (err)
+                reject(err);
+            else
+                resolve(result.rows[0]);
+        });
+    });
+}
+
+export function insertOne(client : PoolClient, string : string, args : unknown[]) {
+    return new Promise((callback, errback) => {
+        const stringReturingId  = string + ' RETURNING uniqueid';
+        client.query(stringReturingId, args, (err, result) => {
+            if (err) {
+                errback(err);
+                return;
+            }
+            if (result === undefined) {
+                errback(new Error('Row does not have uniqueId'));
+                return;
+            }
+            callback(result.rows);
+        });
+    });
+}
+
+const FIELD_NAMES = {
+    app: ['code', 'state', 'name', 'description'] as const,
+    device: ['state'] as const,
+    channel: ['value'] as const
+};
+
+export class PostgresDatabase implements DB.AbstractDatabase {
+    private _pool : Pool;
+    private _transactionQueue = new WeakMap<PoolClient, Promise<unknown>>();
+
+    constructor(public platform : PostgresPlatform) {
+        this._pool = new Pool(platform.getPostgresConfig());
+    }
+
+    withClient<T>(callback : (client : PoolClient) => Promise<T>) : Promise<T> {
+        return new Promise<T>((resolve, reject) => {
+            this._pool.connect((err, c, release) => {
+                if (err) {
+                    console.error('Error in acquiring client', err.stack);
+                    reject(err);
+                    return;
+                }
+                resolve(callback(c));
+                release();
+            });
+        });
+    }
+
+    withTransaction<T>(transaction : (client : PoolClient) => Promise<T>) : Promise<T> {
+        return this.withClient((client) => {
+            const queue = this._transactionQueue.get(client) || Promise.resolve();
+            return new Promise((callback, errback) => {
+                const newqueue = queue.then(async () => {
+                    await client.query('BEGIN;');
+                    try {
+                        const result = await transaction(client);
+                        await client.query('COMMIT;');
+                        callback(result);
+                    } catch(err) {
+                        console.log(`COMMIT ERROR:${JSON.stringify(err)}`);
+                        client.query('ROLLBACK;', () => {
+                            console.log('Rolling back transaction.');
+                        });
+                        errback(err);
+                    }
+                    // continue with the queue
+                });
+                this._transactionQueue.set(client, newqueue);
+            });
+        });
+    }
+
+    ensureSchema() {
+        // Database is initialized out of band
+        // eslint-disable-next-line no-promise-executor-return
+        return new Promise<void>((resolve, reject) => resolve());
```

Comment:
```suggestion
        return Promise.resolve();
```
(alternatively, make the `ensureSchema()` function async and leave it empty)